controllers/util/predicates.go

```diff
@@ -68,3 +70,36 @@ func (p MetadataChangedPredicate) Update(e event.UpdateEvent) bool {
 
 	return metaChanged
 }
+
+// Name Predicate return a predicate the filter events produced
+// by resources that matches the given name
+func NamePredicate(name string) predicate.Predicate {
+	return predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		return obj.GetName() == name
+	})
+}
+
+func CrdCreateAndDeletePredicate(log *logr.Logger, AvailableCrds map[string]bool, CrdName string) predicate.Predicate {
```

Comment:
The only available Crd that you care about in each invocation of this function is the one named `CrdName`, so there is no need to send all of the Available CRDs into this function. 

```suggestion
func CrdCreateAndDeletePredicate(log *logr.Logger, CrdName string, CrdExists bool) predicate.Predicate {
```